File: cpp/se/IPVersion.cpp
#include "se/IPVersion.hpp"

using namespace std;

// Static member initialisation.
vector<const EnumType *> IPVersion::values;

const IPVersion* IPVersion::IPv4()
{
	static const IPVersion ver("v4");
	return &ver;
}

const IPVersion* IPVersion::IPv6()
{
	static const IPVersion ver("v6");
	return &ver;
}

void IPVersion::init()
{
	IPv4();
	IPv6();
}

IPVersion::IPVersion (const string &ip)
    : EnumType (values, ip)
{ }

const IPVersion* IPVersion::value (const string &name)
{
    return (const IPVersion *)EnumType::value (name, values);
}

File: src/main/java/com/ausregistry/jtoolkit2/se/IPVersion.java
package com.ausregistry.jtoolkit2.se;

/**
 * Enumeration of Internet Protocol versions supported by EPP.
 */
public enum IPVersion {
    IPv4("v4"), IPv6("v6");

    private String name;

    IPVersion(String versionName) {
        name = versionName;
    }

    public String toString() {
        return name;
    }

    public static IPVersion value(String name) {
        if (name == null) {
            return IPVersion.IPv4;
        }

        return valueOf("IP" + name);
    }
}

